screeps-game-api/src/lib.rs

```diff
@@ -11,13 +11,71 @@ extern crate serde_derive;
 #[macro_use]
 extern crate stdweb;
 
+/*  Used to get data from a javascript reference back into rust code. 
+
+Macro syntax (`$name` are expressions): 
+js_unwrap!($jsExpr)
+
+For reference, `js!()` is a macro that returns a `stdweb::Value` enum.
+https://docs.rs/stdweb/0.4.8/stdweb/enum.Value.html
+
+Here, `js_unwrap!()` takes any valid javascript expression (expresses a value) 
+and will attempt conversion to the receiving variable type using `try_into`. 
+For example:
+```
+let s: u32 = js_unwrap!(Game.time);
+```
+
+This will be be converted to 
+```
+let s: u32 = js!(Game.time).try_into().expect('Some Err Msg');
+```
+
+Since `Game.time` returns a javascript `number`, `js!` spits out a
+`stdweb::Value::Number` which is convertible to a u32 and should work without
+problem.
+
+A non-exhaustive list of types that work (use your judgement)
+
+  js      |  rust
+------------------------
+Number    | u32, i32, f32
+String    | String, &str
```

Comment:
I don't think `&str` is usable as a result of `js_unwrap!()` since it's a borrowed value and rust has to own the memory it is borrowed from.